src/Paket.Core/Dependencies/PackageResolver.fs

```diff
@@ -946,6 +946,10 @@ let WorkerCount = 6
 
 type PreferredVersionsFunc = ResolverStrategy -> GetPackageVersionsParameters -> list<SemVerInfo * PackageSource list>
 
+type private StepResult =
```

Comment:
I used `StepResult` as return type to avoid recursive call in `step` and `fuseConflicts` functions.